README.md

```diff
@@ -29,20 +29,111 @@ The <b>Agglayer</b> (<i>Aggregation layer</i>) provides a common language for se
+| agglayer-clock                 | Defines the pace of the Agglayer in terms of epoch with support for two clocks: time (for testing) and block (for listening for L1 blocks) |
+| agglayer-config                | Manages configuration settings and parameters for Agglayer components                                       |
+| agglayer-contracts             | Contains smart contracts and related logic                                                                  |
+| agglayer-gcp-kms               | Provides integration with GCP's Key Management Service for secure key handling                              |
+| agglayer-node                  | Responsible for spawning and running the different components of the node                                   |
+| agglayer-prover-types          | Defines data structures and types used by the prover                                                        |
+| agglayer-prover                | Responsible for running everything related to the prover                                                    |
+| agglayer-signer                | Manages signing operations                                                                                  |
+| agglayer-storage               | Contains two layers: a physical layer for abstracting RocksDB and a logic layer for exposing the interface to other crates so that they may interact with the storage |
+| agglayer-telemetry             | Handles telemetry and monitoring functionalities                                                            |
+| agglayer-types                 | Defines common data types and structures                                                                    |
+| agglayer                       | The CLI for interacting with the Agglayer                                                                   |
+| pessimistic-proof-program      | Implements the pessimistic proof program                                                                    |
+| pessimistic-proof-test-suite   | Provides a suite of tests for validating the functionality of the pessimistic proof program                  |
+| pessimistic-proof              | Contains the core logic and implementation of the pessimistic proof mechanism                               |
+
+## Prerequisite
+
+Before working with the repository, you’ll need the following:
+
+### Succinct Prover Network
+
+You’ll need to submit a unique Ethereum address to Succinct for access to their proving network. To get access:
+
+1. Follow the instructions [here](https://docs.succinct.xyz/docs/generating-proofs/prover-network/key-setup) to use Foundry to generate a new private key or retrieve an existing one
+2. Apply for access for the public address associated with your private key to Succinct Network [here.](https://docs.google.com/forms/d/e/1FAIpQLSd-X9uH7G0bvXH_kjptnQtNil8L4dumrVPpFE4t8Ci1XT1GaQ/viewform)
```

Comment:
```suggestion
1. Follow the instructions [here](https://docs.succinct.xyz/docs/generating-proofs/prover-network/key-setup) to use Foundry to generate a new private key or retrieve an existing one.
2. Apply for access for the public address associated with your private key to Succinct Network [here](https://docs.google.com/forms/d/e/1FAIpQLSd-X9uH7G0bvXH_kjptnQtNil8L4dumrVPpFE4t8Ci1XT1GaQ/viewform).
```

Reply:
Also on other places, end of sentence punctuation should not be part of the hyperlink.